torch/lib/c10d/test/ProcessGroupGlooTest.cpp

```diff
@@ -426,6 +426,43 @@ void testBarrier(const std::string& path) {
           std::move(event_lists), GLOO_STR, size, allShapes, /* verify_shapes */ false);
 }
 
+void testMonitoredBarrier(const std::string& path) {
```

Comment:
Why do we have a cpp test instead of just a python test?

Reply:
We could test it all through python, but I figured it's better to have a cpp test for better coverage and since we are planning to call this API from C++ directly once it is part of processgroup wrapper.